WIP: [appveyor] Test Boost 1.62+ only

chwarr opened the pull request:
This is part 1 of 2 in the process of dropping support for versions of
Boost older than two years. Boost 1.61+ or newer is now required. (Bond
will likely continue to work with older versions of Boost, but we'll no
longer actively test against them.)

We'd like to test with the current lowest version of Boost that Bond
supports, Boost 1.61. However, there is no Boost 1.61 available on
AppVeyor, so we start with the lowest version that is available in
AppVeyor and supported by Bond: 1.62.

This AppVeyor change needs to be made before the CMake minimum version
bump that we're planning to make with the Travis CI and README changes.

This is the AppVeyor-side change for
https://github.com/Microsoft/bond/issues/771

ara-ayvazyan commented:
Should we remove the check at [appveyor.yml#L97](https://github.com/Microsoft/bond/blob/master/appveyor.yml#L97)?

ara-ayvazyan commented on appveyor.yml:
```diff
@@ -43,14 +43,14 @@
               BOND_VS_NUM: 12
               BOND_ARCH: 32
-              BOND_BOOST: 58
+              BOND_BOOST: 61
```
Appveyor does not seem to have 1.61 according to https://www.appveyor.com/docs/build-environment/#boost.

ara-ayvazyan replied:
Appveyor does not seem to have 1.61 according to https://www.appveyor.com/docs/build-environment/#boost.

chwarr commented:
I did a force push, as I needed to update the commit message.

Changes since most recent reviews:

* Rewrote Boost selection to be more table-driven.
* Switched to 1.62 as there is no 1.61.

chwarr commented:
I need to so some debugging of this...

chwarr commented:
AppVeyor only has Boost 1.56 and 1.58 binaries for MSVC12...

ara-ayvazyan commented:
https://www.nuget.org/packages/boost-vc120

chwarr commented:
Closing this PR. I'll need to rework how we build with AppVeyor and MSVC 2013 and 2017...
